app/src/main/java/com/duckduckgo/app/home/HomeActivity.kt

```diff
@@ -16,21 +16,39 @@
 
 package com.duckduckgo.app.home
 
+import android.content.Intent
 import android.os.Bundle
 import android.support.v4.app.ActivityOptionsCompat
 import android.support.v7.app.AppCompatActivity
 import com.duckduckgo.app.browser.BrowserActivity
 import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.global.intentText
 import kotlinx.android.synthetic.main.content_home.*
 
 class HomeActivity : AppCompatActivity() {
 
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_home)
 
         searchInputBox.setOnClickListener { showSearchActivity() }
+
+        if (savedInstanceState == null) {
+            consumeSharedText(intent)
+        }
+    }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        consumeSharedText(intent)
+    }
+
+    private fun consumeSharedText(intent: Intent?) {
+        val sharedText = intent?.intentText
```

Comment:
Optional: I think we should start to embrace the Elvis operator, ahuh-huh. ;-) 

This line could be written as:

`val sharedText = intent?.intentText ?: return`

and then the if statement below can be removed.

Reply:
Love an excuse to use the elvis operator 👍